cmd/admin/cert.go

```diff
@@ -221,6 +224,23 @@ func (a *admin) serialsFromRegID(ctx context.Context, regID int64) ([]string, er
 	return serials, nil
 }
 
+func (a *admin) checkValidSerial(serial string) (string, error) {
+	serialStrip := func(r rune) rune {
+		switch {
+		case unicode.IsLetter(r):
+			return r
+		case unicode.IsDigit(r):
+			return r
+		}
+		return rune(-1)
+	}
+	strippedSerial := strings.Map(serialStrip, serial)
+	if !core.ValidSerial(strippedSerial) {
+		return strippedSerial, errors.New("invalidFormat")
```

Comment:
Returning both a value and an error is non-idiomatic Go: if there is an error, we should be returning the zero-value alongside it.
```suggestion
		return "", errors.New("invalid serial format")
```